homeassistant/components/amazon_polly/tts.py

```diff
@@ -151,11 +69,15 @@
 )
 
 
-def get_engine(hass, config, discovery_info=None):
+def get_engine(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> Provider | None:
     """Set up Amazon Polly speech component."""
     output_format = config[CONF_OUTPUT_FORMAT]
     sample_rate = config.get(CONF_SAMPLE_RATE, DEFAULT_SAMPLE_RATES[output_format])
-    if sample_rate not in SUPPORTED_SAMPLE_RATES_MAP.get(output_format):
+    if sample_rate not in SUPPORTED_SAMPLE_RATES_MAP[output_format]:
         _LOGGER.error(
             "%s is not a valid sample rate for %s", sample_rate, output_format
         )
```

Comment:
```python
profile: str | None = config.get(CONF_PROFILE_NAME)
```

Reply:
It's possible to define exact type, but I don't think it's very useful. But str keys are important because they'll be used for `boto3.client` call.
```python
aws_config: dict[str, Any] = {
```